Car_owners/application/views.py

```diff
@@ -1,108 +1,151 @@
+            return response
+
+        # Additional custom response, when no object found
+        if not serializer.data:
+            return Response(f"There is no {self.model_class_name} with given data")
+
+        return Response(serializer.data)
 
+
+@method_decorator(name="retrieve", decorator=swagger_decorator_owner)
+@method_decorator(name="update", decorator=swagger_decorator_owner)
+@method_decorator(name="partial_update", decorator=swagger_decorator_owner)
+@method_decorator(name="destroy", decorator=swagger_decorator_owner)
+class OwnerViewSet(BaseViewSet):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
+        self.queryset = Owner.objects.all()
+        self.serializer_class = OwnerSerializer
+        self.filterset_class = OwnerFilter
+        self.ordering_fields = ["name", "surname"]
         self.model_class = Owner
         self.model_class_name = self.model_class._meta.object_name
-        self.queryset = self.model_class.objects.all()
-        self.case_insensitive_fields = ["name", "surname"]
-
-    def additional_validation_check(
-        self, request_data_dict: Dict[str, str]
-    ) -> (bool, response_type):
-        phone_number = request_data_dict.get("phone", "000000000")
-        if len(phone_number) < 9:
-            return True, Response({"Phone number is to short."})
-        elif len(phone_number) > 9:
-            return True, Response({"Phone number is to long."})
-        else:
-            return False, Response({""})
+
+    def request_validation(self, request: request_type) -> (bool, response_type):
+        for key, value in request.query_params.items():
+            if key == "phone":
+                if search("[^0-9]", value):
+                    return True, Response(
+                        {"phone": "Phone number can contain only digits"}
+                    )
+                elif len(value) > 9:
+                    return True, Response({"phone": "Phone number is too long"})
+                elif len(value) < 9:
+                    return True, Response({"phone": "Phone number is too short"})
+            elif key == "name" or key == "surname":
+                if search("[^A-Z-a-zżźćńółęąśŻŹĆĄŚĘŁÓŃ]", value):
+                    return True, Response(
+                        {
+                            f"{key}": f"{key} can contain only letters and '-' without "
+                            f"whitespaces"
+                        }
+                    )
+            elif key == "ordering":
+                if value not in self.ordering_fields:
+                    ord_fields_string = ", ".join(self.ordering_fields)
+                    return True, Response(
+                        {
+                            "ordering": f"Ordering should be one of the following: "
+                            f"{ord_fields_string}"
+                        }
+                    )
+
+        return False, Response({""})
```

Comment:
you can just return None, and for failed validation return a `Response`. That way you can easily just check if method returned anything, and if it does - it means validation failed